File: 240311/겹치지 않는 선분 2/line-segments-that-do-not-overlap-2.py
n=int(input())

coord=[list(map(int,input().split())) for _ in range(n)]

ans=0

for i in range(n):
    flag=1
    for j in range(n):
        if i==j:
            continue
        if (coord[i][0]<=coord[j][0] and coord[i][1]>=coord[j][1]) or (coord[i][0]>=coord[j][0] and coord[i][1]<= coord[j][1]):
            flag=0
            break
    if flag==1:
        ans+=1

print(ans)

File: 260118/겹치지 않는 선분 2/line-segments-that-do-not-overlap-2.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');
const n = Number(input[0]);
const segments = input.slice(1, n + 1).map(line => line.split(' ').map(Number));
// Please Write your code here.

let ans = 0;

for(let i=0; i<n; i++){
    let isValid = true;
    for(let j=0; j<n; j++){
        if(i===j) continue;
        const [x1, x2] = segments[i];
        const [x3, x4] = segments[j];

        if((x1<=x3 && x4<=x2) || (x3<=x1 && x2<=x4)){
            isValid = false;
            break;
        }
    }
    if(isValid){
        ans++;
    }
}

console.log(ans);